daemon/graphdriver/quota/projectquota.go

```diff
@@ -194,6 +221,24 @@ func setProjectQuota(backingFsBlockDev string, projectID uint32, quota Quota) er
 	return nil
 }
 
+// getQuotaStat - get the quota stat
+// return 2 means quota is on
+func getQuotaStat(backingFsBlockDev string) (int, error) {
```

Comment:
@coolljt0725 getQuotaStat() is a nice addition, but it has nothing to do with adding ext4 project quota support, does it? The code for xfs project quota could just as well not be really setting project quota limits.
So if you wish to add this check, please do that in a separate commit.

This check may be important for both ext4/xfs project quota, however! it is not enough to verify that ext4 project quota can be applied to overlayfs upper layer.

There is a bug in overlayfs since kernel v4.8 that will cause overlayfs to ignore ext4 project quota limits.
The bug is fixed in kernel v4.12 and has been applied to stable kernel 4.9.y (see https://marc.info/?l=linux-unionfs&m=150095448912473&w=2), but ithe bug may very well be on deployed systems (e.g. Ubuntu 16.10).

As far as I know, the only way to check if the bug exists is to set a test hard quota limit (either user/group/project) on overlayfs upper layer and try to write beyond the limit. And if you are going to implement that test, maybe there is no need to bother with the weaker getQuotaStat() test ???

Reply:
@coolljt0725 I second @amir73il opinion that it's better to have a run-time check if project quota is working. Please let me know if you are going to work on this.